### ISSUE TITLE
Autentication type + category updates

### ISSUE THREAD

**Issue by Vlod23** [allegro/allegro-api#9716]
Hello, I have a task to implement Allegro API into a custom e-shop build with .NET (domivosport.sk) and I would like you to answer the following questions:

1. The client would like to start with the following tasks for Slovak market:

   - Export products (create offers) from the e-shop to Allegro AUTOMATICALLY.
   - Import new orders to the e-shop from Allegro AUTOMATICALLY (e.g. once per day).
   - Import parcels/labels to the e-shop from Allegro AUTOMATICALLY (e.g. once per day).
   - Fulfill the orders in the e-shop and export the changes (e.g. order state) to Allegro MANUALLY by clicking a button in the e-shop.

    Considering all these cases, could you please advise on what would be the best solution when it comes to **authentication**? Would it be possible to avoid any manual authentication/confirmation on the user‘s side (so we can automatize the process as much as possible for him)? If it is possible, which authentication method should I choose when registering the application? Is it „device-code“ grant type (even if the app is a web app)? 

2. I understand that the tree of **categories can be updated** frequently. But can all category data change (including category ID) or does the category ID always remain the same for an existing category? My point is: Am I able to use the Allegro category IDs to set up a fixed matching of Allegro categories with e-shop’s categories (so Allegro category updates don't mess up the matching setup)?

Thank you in advance!

**Comment by MaciejFrackowiak** [comment]
1. You should use a [code](https://developer.allegro.pl/tutorials/authentication-and-authorization-m09BlVyo7iY#authorization-code-flow) or [device](https://developer.allegro.pl/tutorials/authentication-and-authorization-m09BlVyo7iY#device-flow) flow authorization - both require user interaction to confirm consent. 
2. We cannot guarantee that category ID will not change - sometimes categories are merged, so their IDs change.

**Comment by Vlod23** [comment]
Maciej, thank you for your reply. I see that the full automatization as our client requests is probably not possible and there will be some user actions needed.

Regarding the category updates, if there is not a single detail I could use for the Allegro category identification in the long-term, I'm wondering how it is possible to achieve correct pairing of stable e-shop categories with changing Allegro categories. Is it expected to check for the Allegro category updates regularly and always update the application code (category IDs)? In another thread, I read that the updates are irregular, so even scheduling such maintenance seems like a problem.
Could you advise what is the expected way of handling this?

**Comment by MaciejFrackowiak** [comment]
Yes, you should check the Allegro category tree as often as it is possible. And yes, only main category changes are scheduled but the rest may appear every single day.
